src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs

```diff
@@ -499,11 +499,15 @@ private async Task CreateInternal(BlittableJsonReaderObject bjro, SubscriptionCr
             using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext serverContext))
             using (serverContext.OpenReadTransaction())
             {
-                // need to wait on the relevant remote node
-                var node = Database.SubscriptionStorage.GetResponsibleNode(serverContext, name);
-                if (node != null && node != ServerStore.NodeTag)
+                var topology = ServerStore.GetClusterTopology(serverContext);
+                if (topology.AllNodes.Count != 1)
                 {
-                    await WaitForExecutionOnSpecificNode(serverContext, ServerStore.GetClusterTopology(serverContext), node, subscriptionId);
+                    // need to wait on the relevant remote node
+                    var node = Database.SubscriptionStorage.GetResponsibleNode(serverContext, name);
+                    if (node != null && node != ServerStore.NodeTag)
+                    {
+                        await WaitForExecutionOnSpecificNode(serverContext, topology, node, subscriptionId);
```

Comment:
Don't we need at least to execute `await ServerStore.Cluster.WaitForIndexNotification(index);` ?

Reply:
no if its single node we waited inside PutSubscription